### PR TITLE
Remove DebugTypeInfo::isImplicitlyIndirect()

### PR DESCRIPTION
**Pull request by adrian-prantl**
I found the corresponding code in LLDB that depended on this hack and
am now removing both. This makes it possible to share the same code
path for top-level archetypes and member types.

rdar://problem/45462765

### CODE REVIEW

**Comment by adrian-prantl** [comment]
@swift-ci test with https://github.com/apple/swift-lldb/pull/1102

**Comment by adrian-prantl** [comment]
@swift-ci test

**Review by dcci — APPROVED** [review]

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift-lldb/pull/1102
@swift-ci test

**Comment by adrian-prantl** [comment]
@bob-wilson This change is necessary to fix printing tuples of generic values in LLDB (see rdar://problem/45462765). It only affects debug info, not codegen, and thus has no ABI implications.

**Comment by adrian-prantl** [comment]
test with https://github.com/apple/swift-lldb/pull/1102
@swift-ci test

**Comment by slavapestov** [comment]
Thank you so much for ripping this out!

**Comment by bob-wilson** [comment]
This looks fine but unless it's really blocking something, let's wait until next week to merge it.

**Review by bob-wilson — APPROVED** [review]

**Pull request merged**
